rfcs/proposed/task_group_dynamic_dependencies/extended_semantics.md

```diff
@@ -0,0 +1,1515 @@
+# API and semantics details for task group dynamic dependencies
+
+*Note:* This document is a sub-RFC of the [umbrella RFC for task group dynamic dependencies](README.md). 
+
+## Table of contents
```

Comment:
I suggest to structure the document a bit different - that way it would be easier for me to understand and review.

- Introduction
- Make 2,3,4 subsections of a "Design" section
- Alternative approaches & "removed functionality"
- API proposal
- Potential enhancements
- Exit criteria & open questions
- Advanced examples

Reply:
Done